scylla-cdc-replicator/src/replication_tests.rs

```diff
@@ -0,0 +1,272 @@
+                ),
+                (),
+            )
+            .await?;
+
+        session.refresh_metadata().await?;
+
+        Ok(())
+    }
+
+    async fn execute_queries(
+        session: &Session,
+        operations: Vec<TestOperation<'_>>,
+    ) -> anyhow::Result<()> {
+        session.use_keyspace("test_src", false).await?;
+        for operation in operations {
+            session.query(operation.0, operation.1).await?;
+        }
+
+        Ok(())
+    }
+
+    async fn replicate(session: &Arc<Session>, name: &str) -> anyhow::Result<()> {
+        let result = session
+            .query(
+                format!("SELECT * FROM test_src.{}_scylla_cdc_log", name),
+                (),
+            )
+            .await?;
+
+        let table_schema = session
+            .get_cluster_data()
+            .get_keyspace_info()
+            .get("test_dst")
+            .ok_or(anyhow!("Keyspace not found"))?
+            .tables
+            .get(&name.to_ascii_lowercase())
+            .ok_or(anyhow!("Table not found"))?
+            .clone();
+
+        let mut consumer = ReplicatorConsumer::new(
+            session.clone(),
+            "test_dst".to_string(),
+            name.to_string(),
+            table_schema,
+        )
+        .await;
+
+        let schema = CDCRowSchema::new(&result.col_specs);
+
+        for log in result.rows.unwrap_or(vec![]) {
+            consumer.consume_cdc(CDCRow::from_row(log, &schema)).await;
+        }
+
+        Ok(())
+    }
+
+    fn fail_test(
+        table_name: &str,
+        original_rows: &[Row],
+        replicated_rows: &[Row],
+        failure_reason: FailureReason,
+    ) {
+        eprintln!("Replication test for table {} failed.", table_name);
+        eprint!("Failure reason: ");
+        match failure_reason {
+            FailureReason::WrongRowsCount(o, r) => eprintln!(
+                "Number of rows not matching. Original table: {} rows, replicated table: {} rows.",
+                o, r
+            ),
+            FailureReason::RowNotMatching(n) => eprintln!("Row {} is not equal in both tables.", n),
+        }
+
+        eprintln!("ORIGINAL TABLE:");
+        for (i, row) in original_rows.iter().enumerate() {
+            eprintln!("Row {}: {:?}", i + 1, row.columns);
+        }
+
+        eprintln!("REPLICATED TABLE:");
+        for (i, row) in replicated_rows.iter().enumerate() {
+            eprintln!("Row {}: {:?}", i + 1, row.columns);
+        }
+
+        panic!()
+    }
+
+    async fn compare_changes(session: &Session, name: &str) -> anyhow::Result<()> {
+        let original_rows = session
+            .query(format!("SELECT * FROM test_src.{}", name), ())
+            .await?
+            .rows
+            .unwrap_or(vec![]);
+        let replicated_rows = session
+            .query(format!("SELECT * FROM test_dst.{}", name), ())
+            .await?
+            .rows
+            .unwrap_or(vec![]);
```

Comment:
It would be even better for the test to also compare timestamps of at least one column
(you can select the timestamp of a particular column using CQL `writetime(...)` function)

Reply:
This was meant to be here, but we wanted to merge this PR asap, so we decided to delay this: #31